cpp/graph_algorithms_module/algorithms/bipartite_matching.cpp

```diff
@@ -0,0 +1,138 @@
+#include <bits/stdc++.h>
+
+#include <queue>
+
+#include "algorithms/algorithms.hpp"
+
+namespace {
+size_t dfs(uint32_t node, const std::vector<std::vector<uint32_t>> &adj_list,
+           std::vector<bool> *visited, std::vector<uint32_t> *matched) {
+  if ((*visited)[node]) return 0;
+  (*visited)[node] = true;
+  for (uint32_t nxt : adj_list[node]) {
+    if ((*matched)[nxt] == 0 ||
+        dfs((*matched)[nxt], adj_list, visited, matched)) {
+      (*matched)[nxt] = node;
+      return 1;
+    }
+  }
+  return 0;
+}
+
+/// The method runs a BFS algorithm and colors the graph in 2 colors.
+/// The graph is bipartite if for any vertex all neighbors have
+/// the opposite color. In that case, method will return true.
+/// Otherwise, false.
+bool IsSubgraphBipartite(const graphdata::GraphView &G,
+                         std::vector<uint32_t> &colors, uint32_t node_index) {
+  std::queue<uint32_t> unvisited;
+  unvisited.push(node_index);
+  colors[node_index] = 1;
+
+  while (!unvisited.empty()) {
+    uint32_t current_index = unvisited.front();
+    unvisited.pop();
+    for (const auto &neighbour : G.Neighbours(current_index)) {
+      if (neighbour.node_id == current_index) {
+        // Self loops are not allowed
+        return false;
+      }
+      if (colors[neighbour.node_id] == colors[current_index]) {
+        return false;
+      }
+      if (colors[neighbour.node_id] == -1) {
+        colors[neighbour.node_id] = 1 - colors[current_index];
```

Comment:
We are comparing unsigned with signed, and subtracting from 1. 
@dsantl why aren't colors int32_t?